imports/methods/announcements/post.ts

```diff
@@ -0,0 +1,38 @@
+import { check } from 'meteor/check';
+import { ValidatedMethod } from 'meteor/mdg:validated-method';
+import { Meteor } from 'meteor/meteor';
+import Ansible from '../../Ansible';
+import Announcements from '../../lib/models/Announcements';
+import MeteorUsers from '../../lib/models/MeteorUsers';
+import PendingAnnouncements from '../../lib/models/PendingAnnouncements';
+import { userMayAddAnnouncementToHunt } from '../../lib/permission_stubs';
+
+export default new ValidatedMethod({
+  name: 'Announcements.methods.post',
+
+  validate({ huntId, message }: { huntId: unknown, message: unknown }) {
+    check(this.userId, String);
+    check(huntId, String);
+    check(message, String);
+
+    if (!userMayAddAnnouncementToHunt(this.userId, huntId)) {
```

Comment:
So we have a number of slightly-more-complicated situations where we might need to:

* validate types of arguments
* retrieve an object or two from the database (puzzle/hunt)
* consider a field from that object and reject the request based on the current hunt configuration
* do something else based on other fields from the object

(example: [addGuessForPuzzle](https://github.com/deathandmayhem/jolly-roger/blob/main/imports/server/guesses.ts#L151-L197), where we fetch Puzzle, Hunt, and then consult `hasGuessQueue`)

Which of these checks would you imagine belong in `validate` vs in `run`?  If we expect all of the "client sent us something wrong" error classes to surface in `validate`, then we might have to fetch things (example: `Puzzle`) from the DB twice, which feels a little silly and permits TOCTOU (but we probably do that in a number of places anyway).  If the checking can be split between the methods, I wonder how much value we're actually getting from the validate/run split.

Reply:
FWIW in this revision, I'm taking the stance that only type validation of the parameters happens in `validate` and shifted the permission check into `run`.

Reply:
That seems like the easiest thing to keep consistent. 👍🏻